src/lib/Form/DataTransformer/ContentInfoTransformer.php

```diff
@@ -9,38 +9,25 @@
 namespace EzSystems\EzPlatformAdminUi\Form\DataTransformer;
 
 use eZ\Publish\API\Repository\ContentService;
-use eZ\Publish\API\Repository\Exceptions\UnauthorizedException;
-use Symfony\Component\Form\DataTransformerInterface;
 use eZ\Publish\API\Repository\Values\Content\ContentInfo;
-use Symfony\Component\Form\Exception\TransformationFailedException;
 use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
 
 /**
  * Transforms between a Content's ID and a domain specific ContentInfo object.
  */
-class ContentInfoTransformer implements DataTransformerInterface
+final class ContentInfoTransformer implements DataTransformerInterface
 {
-    /** @var ContentService */
-    protected $contentService;
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    private $contentService;
 
-    /**
-     * @param ContentService $contentService
-     */
     public function __construct(ContentService $contentService)
     {
         $this->contentService = $contentService;
     }
 
-    /**
-     * Transforms a domain specific ContentInfo object into a Content's ID.
-     *
-     * @param ContentInfo|null $value
-     *
-     * @return mixed|null
-     *
-     * @throws TransformationFailedException
-     */
-    public function transform($value)
+    public function transform($value): ?int
```

Comment:
Cannot drop PHPDoc w/o making the prototype strict (not saying that you have to make it now).

Reply:
@alongosz Adding type specification to `$value` parameter is not an option here. `\EzSystems\EzPlatformAdminUi\Form\DataTransformer\ContentInfoTransformer::transform` needs to be compatible with interface (otherwise "Declaration must be compatible with DataTransformerInterface->transform(value)").

Reply:
PHPDoc restored in https://github.com/ezsystems/ezplatform-admin-ui/pull/1148/commits/42437d627f79cccd4225b508580a7512e00f0d8d